src/widget/wsearchlineedit.cpp

```diff
@@ -388,12 +389,16 @@ void WSearchLineEdit::keyPressEvent(QKeyEvent* keyEvent) {
         break;
     case Qt::Key_Left:
     case Qt::Key_Right: {
+        // Both keys may change or clear the selection (suggested completion).
         QLineEdit* pEdit = lineEdit();
-        bool hadSelectedText = pEdit->hasSelectedText();
+        bool hadSelectedText = pEdit ? pEdit->hasSelectedText() : false;
```

Comment:
These null checks do not help and only complicate the code. We expect that there is always a line QLineEdit. Just invoke `lineEdit()` without using any mutable, local variables.

Reply:
There is no guarantee that line edit is not null: 
https://github.com/qt/qtbase/blob/e0a5915f88a780ba0791bce5ed042e92036a4900/src/widgets/widgets/qcombobox.cpp#L1697
Better safe than sorry.

Reply:
nup, I figured it's the other way around:
if it's editable then there's always a QLineEdit.
And since the editable state is not changed there will always be a lineEdit()
https://github.com/mixxxdj/mixxx/pull/11289/files#diff-85c225cd63139457cde84ddb0d2f69c854c16f8254672cd19587fd25dd755c62L94

Reply:
Then you need to use VERIFY_OR_DEBUG_ASSERT

Reply:
For sake of completeness, if `isEditable` is true then there is a QLineEdit, otherwise lineEdit() is a nullptr:
```
void QComboBox::setEditable(bool editable)
{
    Q_D(QComboBox);
    if (isEditable() == editable)
        return;

    QStyleOptionComboBox opt;
    initStyleOption(&opt);
    if (editable) {
        if (style()->styleHint(QStyle::SH_ComboBox_Popup, &opt, this)) {
            d->viewContainer()->updateScrollers();
            view()->setVerticalScrollBarPolicy(Qt::ScrollBarAsNeeded);
        }
        QLineEdit *le = new QLineEdit(this);
        le->setPalette(palette());
        setLineEdit(le);
    } else {
        if (style()->styleHint(QStyle::SH_ComboBox_Popup, &opt, this)) {
            d->viewContainer()->updateScrollers();
            view()->setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
        }
        setAttribute(Qt::WA_InputMethodEnabled, false);
        d->lineEdit->hide();
        d->lineEdit->deleteLater();
        d->lineEdit = nullptr;
    }

    d->updateDelegate();
    d->updateFocusPolicy();

    d->viewContainer()->updateTopBottomMargin();
    if (!testAttribute(Qt::WA_Resized))
        adjustSize();
}
```